atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraServersConfigs.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cassandra;
+
+
+import java.net.InetSocketAddress;
+import java.util.Set;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.collect.ImmutableSet;
+
+public final class CassandraServersConfigs {
+    protected CassandraServersConfigs() {
+
+    }
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type",
+            defaultImpl = DefaultCassandraServersCqlDisabledConfig.class)
+    @JsonSubTypes(
+            {
+                    @JsonSubTypes.Type(value = DefaultCassandraServersCqlDisabledConfig.class,
+                            name = DefaultCassandraServersCqlDisabledConfig.TYPE),
+                    @JsonSubTypes.Type(value = ImmutableExplicitCassandraServersCqlDisabledConfig.class,
+                            name = ExplicitCassandraServersCqlDisabledConfig.TYPE),
+                    @JsonSubTypes.Type(value = ImmutableCassandraServersCqlEnabledConfig.class,
+                            name = CassandraServersCqlEnabledConfig.TYPE)
+            }
+    )
+    public interface CassandraServersConfig {
+        Set<InetSocketAddress> thrift();
+        Set<InetSocketAddress> cql();
+        String type();
+    }
+
+    @JsonDeserialize(as = DefaultCassandraServersCqlDisabledConfig.class)
+    @JsonTypeName(DefaultCassandraServersCqlDisabledConfig.TYPE)
+    public static class DefaultCassandraServersCqlDisabledConfig implements CassandraServersConfig {
+        public static final String TYPE = "default";
+
+        private Set<InetSocketAddress> thriftServers;
+
+        @JsonCreator
+        public DefaultCassandraServersCqlDisabledConfig(InetSocketAddress... thriftServers) {
```

Comment:
these names are quite long, given that we have to further qualify it with `CassandraServersConfig` it makes it quite unreadable/annoying to type in tests and other usages as we can see below.

It might be worth adding some static factory methods *inside* `CassandraServersConfig` to make it easier to consumer. Or you could make all three impls in their own file, that said I quite like what you've done here and I bias towards the former.

e.g.

```java
public class CassandraServersConfig {
  // note, these inner classes can now become private/potentially package private because you can just make them using the top level static factory methods.
  private static class Default... { ... }

  public static CassandraServersConfig thriftOnlyConfig(InetSocketAddress... thriftServers) {
    return new Default... (thriftServers);
  }
}

Reply:
Shortened names of the specific config classes as the should be prefixed by `CassandraServersConfigs. <ConfigClass>` (can revisit this if it is non standard) and added factory methods.